transcendence-app/src/user/user.controller.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common';
+import { UserService } from './user.service';
+import { UserDto } from './dto/user.dto';
+
+@Controller('user')
+export class UserController {
+  constructor(private userService: UserService) {}
+
+  @Post(':id')
+  async addUser(@Param('id') param: string): Promise<UserDto> {
+    const id = Number(param);
+    const user: UserDto = {
+      id: id,
+      provider: 'ft_transcendence',
+      image_url: 'www.img_url.com/hello.jpg',
+      username: `user_${id}`,
+      email: 'afgv@github.com',
+    };
+    this.userService.findOneOrCreate(user);
+    return user;
+  }
+
+  @Get(':id')
+  async getUserById(@Param('id') param: string): Promise<UserDto | undefined> {
+    const result: UserDto | undefined = this.userService.retrieveUserWithId(
+      Number(param),
+    );
+    if (result === undefined) throw new NotFoundException();
```

Comment:
I had to force push because I rebased, and now your comment is gone, but I heeded it.